Sources/SignalRClient/DefaultTransportFactory.swift

```diff
@@ -10,18 +10,64 @@ import Foundation
 
 internal class DefaultTransportFactory: TransportFactory {
     let logger: Logger
-
-    init(logger: Logger) {
+    let permittedTransportTypes: TransportType
+    var orderOfPreference: [TransportType] = [.webSockets, .longPolling]
```

Comment:
This is how the fallback mechanism is implemented - by manipulating the order of this array. I did it this way in part because it is extensible - if someone was to implement the server sent events transport, it could be added to the array here and it should just work properly.

I did try some simpler implementations but this was what I came up with that worked well in my application.

Reply:
No problem. I still may want to simplify it later. I am not planning on implementing or even taking a PR that adds support for the SSE transport. It might have been useful in 2012 but I don't think it's the case anymore. I do see value in having longpolling transport as a fallback though.